### PR TITLE
Samplegen: Java template small fixes

### PR DESCRIPTION
**Pull request by hzyi-google**

### CODE REVIEW

File: src/main/resources/com/google/api/codegen/java/method_sample.snip [hunk 1]
```diff
@@ -80,9 +85,15 @@
 
 @snippet asyncOperationCallableMethodSampleCode(apiMethod, initCode)
   {@initCode(initCode)}
-  OperationFuture<{@apiMethod.callableMethod.genericAwareResponseType}> future = {@sampleFutureMethodCall(apiMethod)};
```
Comment (andreamlin) [inline src/main/resources/com/google/api/codegen/java/method_sample.snip]: Do you know why we used `genericAwareResponseType` in the first place?

Comment (hzyi-google) [reply src/main/resources/com/google/api/codegen/java/method_sample.snip]: It seems it's converting `Empty` to `Void`. For exmaple, the callable methods for those rpcs who return `Empty` would return `ApiFuture<Void>` instead to be more Java idiomatic. But it seems we do not do this conversion for `LongRunningCallable`s.

**Comment by andreamlin** [comment]
I think @lukesneeringer is working on expanding the `run_generated_tests` CI target. After that, we can merge if the only thing failing is the artman_smoke_tests (this has a preexisting problem).

**Comment by hzyi-google** [comment]
Cool. They seem to be running now. Thanks for the heads up.

**Comment by hzyi-google** [comment]
@andreamlin @lukesneeringer Would there be any permission issue to be sorted out for CI? The test has been <i>in progress<i> for more than an hour.

**Comment by andreamlin** [comment]
@hzyi-google You can merge whenever the button says you can!

**Pull request merged**